prime-router/src/main/kotlin/fhirengine/translation/BundleEnhancer.kt

```diff
@@ -0,0 +1,74 @@
+package gov.cdc.prime.router.fhirengine.translation
+
+import ca.uhn.hl7v2.model.Message
+import gov.cdc.prime.router.fhirengine.utils.HL7Reader
+import io.github.linuxforhealth.hl7.data.Hl7RelatedGeneralUtils
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.DateTimeType
+import org.hl7.fhir.r4.model.Extension
+import org.hl7.fhir.r4.model.Patient
+
+class BundleEnhancer {
```

Comment:
If you make this a non-static class and delegate the run task to the client, it could make this class more:

- testable: If you follow this approach, you could have unit tests for each method in here. But, the way it is now, in order to test these methods you have to test them via `HL7toFhirTranslator` integration tests.
- extensible/cohesive: we would delegate the run task to the client, that way different clients in the future can choose exactly what enhancements they run by easily chaining them together, instead of the enhancements being hardcoded in this class itself

Here is the patch that implements this suggestion (easier then explaining everything). Each enhancer returns `this` so you can chain the calls in the client

[OOP_suggestions.patch](https://github.com/CDCgov/prime-reportstream/files/12729019/OOP_suggestions.patch)

If this makes sense and you choose to do this, then let's also write unit tests for the enhancer methods.

Reply:
I would recommend doing this differently, and not having a class for BundleEnhancer at all, instead using extension functions on Bundle. [This patch](https://github.com/CDCgov/prime-reportstream/files/12729752/Use_extension_functions_for_BundleEnhancer_kt.patch) demonstrates it.
It has the same benefits that Arnej highlights above, but instead of having this object `BundleEnhancer`, we can literally enhance `Bundle` to have these functions.

One slight issue with this approach is that we would have extension functions in both `BundleEnhancer.kt` and `FHIRBundleHelpers.kt`, with the later having a wrapping class. I would personally eliminate the class `FHIRBundleHelpers` and have all of the helping functions from both files in the same file without a class, but might be an overreach for this story, so I'd be okay holding off on that for now.

Reply:
Jimmy's idea makes sense to me and sounds like a good way to better utilize Kotlin features for more maintainable code